### PR TITLE
Decouple category filter modal

### PR DESCRIPTION
**Pull request by MythicManiac**
This PR replaces #879 as some commits in #879 had to be merged ahead of time; this PR is the re-ordered result.

This should be merged after #886

### CODE REVIEW

File: src/components/modals/CategoryFilterModal.vue
Comment (MythicManiac) [inline src/components/modals/CategoryFilterModal.vue:90]: Does this work as intended, as in, do these get cleared from the cached values when appropriate (e.g. game change)?

File: src/components/modals/CategoryFilterModal.vue
Comment (MythicManiac) [inline src/components/modals/CategoryFilterModal.vue:121]: It looks like `this.activeGame` is always set, is there a reason we don't use it directly on line 90?

Though I'd be more comfortable taking in the active game as a prop rather than relying on global state

**Review by MythicManiac — COMMENTED** [review]
A few questions, otherwise looks good. Can't approve this myself since I re-opened the PR on behalf of @anttimaki

File: src/components/modals/CategoryFilterModal.vue
Comment (anttimaki) [reply src/components/modals/CategoryFilterModal.vue:90]: This approach is copied from the Manager page. AFAIK this is just a dummy value so we don't have to check if `settings` is `undefined` in every place where it's called (or use `!` here). The actual value will be overwritten in `created`.

And yes, based on manual testing it works as intended. The whole component will get unmounted/remounted if user wants to change the game. At least for now when the game can only be changed in the `GameSelectionScreen`.

File: src/components/modals/CategoryFilterModal.vue
Comment (anttimaki) [reply src/components/modals/CategoryFilterModal.vue:121]: I don't think we can call `await` on line 90. But since `this.activeGame` is used only on this line, we can call `GameManager.activeGame` on this line and ditch `this.activeGame`.

I've assumed one of the main reasons the static class/singleton pattern is used widely on this code base is that we can access e.g. the active game from anywhere without having to pass it around. I'm not against passing it as a prop, but if we no longer want to access it via GameManager, I'd suggest maybe moving it into Vuex store.

**Review by anttimaki — APPROVED** [review]
Marking approved on behalf of Mythic.

File: src/components/modals/CategoryFilterModal.vue
Comment (ebkr) [inline src/components/modals/CategoryFilterModal.vue:123]: Might be good to link this TODO to something in clickup

File: src/store/modules/ModFilterModule.ts
Comment (ebkr) [inline src/store/modules/ModFilterModule.ts:36]: Rename to `categories`
imo there's no reason to shorten this

**Review by ebkr — APPROVED** [review]
One minor comment and a suggestion to not have a permanent TODO. Happy to merge once resolved.

File: src/components/modals/CategoryFilterModal.vue
Comment (MythicManiac) [reply src/components/modals/CategoryFilterModal.vue:123]: Removed this comment entirely since I don't think there's any way to toggle the theme while this modal is open, but if you feel like this should be tracked as a TODO let me know

Comment (anttimaki) [reply src/components/modals/CategoryFilterModal.vue]: @MythicManiac While the modal isn't open, it's still mounted when user changes the color theme via settings, meaning the value won't get updated until user leaves and returns to the manager page. (And to clarify, the problem existed and was in fact worse before the changes in this PR.)

Also noticed you removed another TODO comment. Is this to say we should avoid TODO comments in this code base?

**Pull request merged**
